doc/registered_filters.md

```diff
@@ -0,0 +1,43 @@
+# Registered Filter Plugins
+
+The filter identifier is designed to be a unique identifier for the filter. Values from zero through 32,767 are reserved for filters supported by The HDF Group in the HDF5 library and for filters requested and supported by the 3rd party.
+
+Values from 32768 to 65535 are reserved for non-distributed uses (e.g., internal company usage) or for application usage when testing a feature. The HDF Group does not track or document the usage of filters with identifiers from this range.
+
+Please contact the maintainer of a filter for help with the plugin.
+
+| WARNING: be aware that compression filters require that the library not use `H5_MEMORY_ALLOC_SANITY_CHECK`.  Building in debug mode automatically enables this feature in earlier releases, which causes memory allocation and free problems in filter applications. Future versions of HDF5 will not enable this feature. |
```

Comment:
I wonder if this should say "current versions" of HDF5 rather than "future versions"